client/src/pages/Dashboard.tsx

```diff
@@ -70,28 +73,112 @@ const Dashboard: React.FC = () => {
+        return acc;
+      }, {});
+      setAgreeCheckboxStates(initialAgreeCheckboxStates);
+
+      const updatedAnswerList = updatedAnswers.map(answer => ({
+        ...answer,
+        // commonsensicality: commonsensicalityScores[answer.statementId] || 0,
+        agreement: agreementPercentages[answer.statementId] || { I_agree: 0, others_agree: 0 },
+      }));
+      setAnswerList(updatedAnswerList);
+
+    } catch (error) {
+      console.log("Error fetching answers:", error);
+    }
+  };
+
   useEffect(() => {
-    const getAnswers = async () => {
-      if (token === null) return null;
+    getAnswers();
+  }, []);
+
+  //  Change I_agree or others_agree variable -- Usage:
+  // For I_agree: handleCheckboxChange(id, 'I_agree')
+  // For others_agree: handleCheckboxChange(id, 'others_agree')
+  const handleCheckboxChange = async (id: number, type: 'I_agree' | 'others_agree') => {
+    const stateKey = type === 'I_agree' ? 'agreeCheckboxStates' : 'checkboxStates';
+    const setState = type === 'I_agree' ? setAgreeCheckboxStates : setCheckboxStates;
+    const newCheckedState = !((type === 'I_agree' ? agreeCheckboxStates : checkboxStates)[id]);
+    const currentAnswer = answerList.find(answer => answer.id === id);
+
+    if (!currentAnswer) {
+      console.error("Answer not found");
+      return;
+    }
+
+    setState((prevStates) => ({
+      ...prevStates,
+      [id]: newCheckedState,
+    }));
+
+    try {
+      Backend.defaults.headers.common["Authorization"] = token;
+      const response = await Backend.post("/answers/changeanswers", {
+        statementId: currentAnswer.statementId,
+        I_agree: type === 'I_agree' ? (newCheckedState ? 1 : 0) : (currentAnswer.I_agree ? 1 : 0),
+        I_agree_reason: "n/a - answer changed",
+        others_agree: type === 'others_agree' ? (newCheckedState ? 1 : 0) : (currentAnswer.others_agree ? 1 : 0),
+        others_agree_reason: "n/a - answer changed",
+        perceived_commonsense: 0,
+        sessionId: surveySession,
+      });
+      console.log(`${type} updated`, response.data);
+    } catch (error) {
+      console.log(`Error updating ${type}`, error);
+      // Revert the state change in case of an error
+      setState((prevStates) => ({
+        ...prevStates,
+        [id]: !newCheckedState,
+      }));
+    }
+  };
+
+  const useEditAnswer = async () => {
+    if (editing) {
+      if (!token) return;
+
       try {
-        Backend.defaults.headers.common["Authorization"] = token;
-        const response = await Backend.post("/answers/getanswers", {
-          email: "user@test.com",
-        });
-        setAnswerList(response.data);
-        return response.data.ok;
+        await getAnswers(); // call getAnswers to refresh the data and get new percentage
       } catch (error) {
-        console.log(error);
+        console.log("Error refreshing data in useEditAnswer:", error);
       }
-    };
-    getAnswers();
-  }, []);
+    }
+
+    setEditing(!editing);
+  };
```

Comment:
I see thank you! I didn't use useEffect() here because the action is called when the user presses the "Done" button. This is mainly because if I connect it to the toggle state (which is dependent on `editing`), if the user toggles too fast the percentages update with a lag and looks a little buggy. So I made it so the percentages change only when the user finalizes their changes and presses "Done" which is a specific user event